### PR TITLE
Added Legend to Onset picker field-set

### LINKED ISSUE #461: Onset Picker does not include legend with fieldset
# Bug Report

## Description
Fieldset used in terra-clinical-onset-picker does not contain a legend element. It is recommended that fieldsets should contain a legend element that provides a description of the field group.

## Steps to Reproduce

1. Navigate to https://engineering.cerner.com/terra-ui/#/components/terra-clinical-onset-picker/clinical-onset-picker/clinical-onset-picker
2. Inspect fieldset rendered by terra-clinical-onset-picker.

## Expected Behavior
Onset picker aligns with accessibility recommendations.

## Possible Solutions
1. Include legend with fieldset. This could include a prop that toggles if it is visually-hidden or not using terra-visually-hidden-text
2. Use a div as a wrapping container instead of a fieldset.

## Environment

* Component Name and Version: terra-clinical-onset-picker
* Browser Name and Version: All

### PR DESCRIPTION
**Pull request by supreethmr**
### Summary
Resolves #461 

### Additional Details
Jest and WDIO Snapshots of onset picker were failing at master branch due to the changes made to it's dependent component like terra-form-select : https://github.com/cerner/terra-core/pull/2236.

Snapshots has been updated to latest code changes of dependent component.

Thanks for contributing to Terra. 
@cerner/terra

### CODE REVIEW

File: packages/terra-clinical-onset-picker/src/OnsetPicker.jsx [hunk 1]
```diff
@@ -456,6 +469,7 @@ class OnsetPicker extends React.Component {
       <div id={this.props.id} {...customProps}>
         <FieldSet className={cx('fieldset')}>
           {/* Precision */}
+          <legend className={cx(['legend', { 'legend-hidden': isLegendHidden }])}>{legend}</legend>
```
Comment (jmsv6d) [inline packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: I think that we should just pass through `legend` and `legendAttrs` props to the `FieldSet`. If we want an `isLegendHidden` prop it should be added to `FieldSet` and just passed through here for consistency.

Comment (supreethmr) [reply packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: Sorry, I didn't get that assuming and replying something might create more confusion to the context you're expecting so can you please elaborate that little.

Comment (nramamurth) [reply packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: +1 to @jmsv6d's comment. The `terra-form-fieldset` component already [does the work](https://github.com/cerner/terra-core/blob/725f3a1b80773dbc8da3353f4acc1a0b518c82a5/packages/terra-form-fieldset/src/Fieldset.jsx#L54-L60) for you, I'd simply pass the prop down.

Comment (supreethmr) [reply packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: I think I might hold this PR until code changes in Terra-core for terra-form-fieldset gets merged to pass isLegendHidden Prop through onset picker.

**Comment by supreethmr** [comment]
Logged new PR: https://github.com/cerner/terra-clinical/pull/489 for these code changes since these code changes had few WDIO snapshots which were not required with new changes.

**Pull request closed without merging**